ApplicationLibCode/ProjectDataModel/Summary/RimEnsembleStatisticsCase.cpp

```diff
@@ -56,83 +53,104 @@ const std::vector<time_t>& RimEnsembleStatisticsCase::timeSteps() const
 //--------------------------------------------------------------------------------------------------
 ///
 //--------------------------------------------------------------------------------------------------
-const std::vector<double>& RimEnsembleStatisticsCase::p10() const
+std::vector<time_t> RimEnsembleStatisticsCase::timeSteps( const RifEclipseSummaryAddress& resultAddress ) const
 {
-    return m_p10Data;
+    return m_timeSteps;
 }
 
 //--------------------------------------------------------------------------------------------------
 ///
 //--------------------------------------------------------------------------------------------------
-const std::vector<double>& RimEnsembleStatisticsCase::p50() const
+bool RimEnsembleStatisticsCase::values( const RifEclipseSummaryAddress& resultAddress, std::vector<double>* values ) const
 {
-    return m_p50Data;
+    const std::vector<double>* sourceData   = nullptr;
+    auto                       quantityName = resultAddress.ensembleStatisticsVectorName();
+
+    if ( quantityName == ENSEMBLE_STAT_P10_QUANTITY_NAME )
+        sourceData = &p10();
+    else if ( quantityName == ENSEMBLE_STAT_P50_QUANTITY_NAME )
+        sourceData = &p50();
+    else if ( quantityName == ENSEMBLE_STAT_P90_QUANTITY_NAME )
+        sourceData = &p90();
+    else if ( quantityName == ENSEMBLE_STAT_MEAN_QUANTITY_NAME )
+        sourceData = &mean();
+
+    if ( !sourceData ) return false;
+
+    values->clear();
+    values->reserve( sourceData->size() );
+    for ( auto val : *sourceData )
+        values->push_back( val );
```

Comment:
Isn't `*values = *sourceData` simpler here?

Reply:
Maybe also consider returning a copy of the data instead of passing a pointer to a std::vector?

Reply:
Refactoring issue https://github.com/OPM/ResInsight/issues/10648